src/coprocessor/endpoint.rs

```diff
@@ -610,10 +610,42 @@ mod tests {
         }
     }
 
+    async fn yield_now() {
+        use std::future::Future;
+        use std::pin::Pin;
+        use std::task::{Context, Poll};
+
+        struct YieldNow {
+            yielded: bool,
+        }
+
+        impl Future for YieldNow {
+            type Output = ();
+
+            fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<()> {
+                if self.yielded {
+                    return Poll::Ready(());
+                }
+
+                self.yielded = true;
+                cx.waker().wake_by_ref();
+                Poll::Pending
+            }
+        }
+
+        YieldNow { yielded: false }.await
+    }
+
     #[async_trait]
     impl RequestHandler for UnaryFixture {
         async fn handle_request(&mut self) -> Result<coppb::Response> {
-            thread::sleep(Duration::from_millis(self.handle_duration_millis));
+            // We split the task into small executions of 1 second.
+            for _ in 0..self.handle_duration_millis / 1_000 {
+                thread::sleep(Duration::from_millis(1_000));
+                yield_now().await;
```

Comment:
This depends on the feature that yatp doesn't actually yield if a future is immediately ready after returning `Pending`. Otherwise https://github.com/tikv/tikv/blob/866dc5e8c6d0c1fda80d5ee96b68752cfb743ceb/src/coprocessor/endpoint.rs#L1230 will fail.

Can we think of a better idea? Maybe we can test wait time and process time in different tests?

Reply:
Maybe we can test them separately.